src/app/search/facet-search/facet-search.pipe.ts

```diff
@@ -0,0 +1,23 @@
+import { Pipe, PipeTransform } from '@angular/core';
+
+@Pipe({
+  name: 'getFacetSearchResults',
+})
+/**
+ * This pipe filters results in the facet according to the facet search text
+ *
+ * @param {Array<any>} items The items in a facet
+ * @param {string} searchText The search text entered
+ * @param {string} facet The facet the search applies to
+ * @returns {Array<any>} The results that match the search
+ * @memberof GetFacetSearchResultsPipe
+ */
+export class GetFacetSearchResultsPipe implements PipeTransform {
+  transform(items: Array<any>, searchText: string, facet: string): Array<any> {
+    if (!items || !searchText || facet !== 'author') {
+      return items;
+    }
+    const filteredItems = items.filter((item) => item.toLowerCase().indexOf(searchText.toLowerCase()) !== -1);
```

Comment:
1. Micro-optimization that maybe the compiler already handles, but you would ideally create a variable out of `searchText.toLowerCase()` -- otherwise you are executing it again and again for every item in the array.
2. You could use `includes` instead of `indexOf` like you do in facet-search-update-pipe